### DISCUSSION TITLE
can you create a rule for rounded corners?

### DISCUSSION THREAD

**Discussion by voziand** [yshui/picom#874] [General]
Is there a way to create specific rule on which windows class should get a specific corner radius?
for example, i want all the windows to have a 12 radius but dwm should only have an 8 corner radius?
i tried 
rounded-corner-rule [
"8:class_g = 'dwm' "
]
no success tough.

**Comment by ohxxm** [comment]
Rounded corners don't even work for me, some have gotten it working but not me. I did make a conversation in this discussion tab about 3 months ago but only 3 upvotes and 0 response.

**Reply by voziand** [reply]
what does your config file look like? maybe i can help
also are you trying to make it work on a vm or physical?

**Reply by ohxxm** [reply]
```
# Animations
animations = true

animation-for-transient-window = "fly-in"
animation-for-open-window = "zoom"
animation-for-unmap-window = "zoom"

animation-dampening = 20
animation-clamping = false

# Corners
corner-radius = 10;
# round-borders = 10;


rounded-corners-exclude = [
    "class_g='Bar'",
    "class_g='Rofi'",
    "class_g='dwm'",
    # "QTILE_INTERNAL:32c = 1",
]

# Blur
blur-method = "dual_kawase";
blur-strength = 8;
blur-background = true;
active-opacity = 1;
inactive-opacity = 1;
inactive-opacity-override = false;

opacity-rule = [
  "85:class_g = 'Alacritty'",
  "90:class_g = 'st-256color'",
]

blur-background-exclude = [
  "window_type = 'dock'",
  "class_g = 'slop'",
  "QTILE_INTERNAL:32c = 1",
  #"window_type = 'desktop'",
  "_GTK_FRAME_EXTENTS@:c",
  "role = 'xborder'",
];


# Fading Windows

fading = true;
fade-delta = 10;
no-fading-openclose = false;

# Other Settings
backend = "glx";
refresh-rate = 165;
unredir-if-possible = true
vsync = true;
```

**Reply by voziand** [reply]
try with vsync = false;

also you are missing -> ; at the end of line 2,4,5,6,8,9,21,34,55. sometimes it doesn't matter but who knows...

**Reply by ohxxm** [reply]
aight i will try this

**Reply by ohxxm** [reply]
doesn't work

**Reply by voziand** [reply]
try picom --corner-radius 10, in your .xinitrc instead of --experiemntal-backends.
instead of 10 put the corner radius you want.

**Reply by absolutelynothelix** [reply]
> Rounded corners don't even work for me, some have gotten it working but not me. I did make a conversation in this discussion tab about 3 months ago but only 3 upvotes and 0 response.

rounded corners should be stable now and work for everyone, if you're having issues with them using picom built from the latest commit in the `next` branch feel free to open an issue.
> ```
> # Animations
> ```

there are no animations in picom currently and we can't help you if you're using a picom's fork.
> also you are missing -> ; at the end of line 2,4,5,6,8,9,21,34,55. sometimes it doesn't matter but who knows...

it shouldn't matter.

**Comment by yshui** [comment]
There is `rouned-corners-exclude`.

**Reply by voziand** [reply]
i know. Burt that exudes the app from having rounded corners all together. What i'm asking is if it is possible to create a rule that will set a different radius of rounded corners on specific windows.  In my case, i want all regular windows to have 16px radius but the DWM bar only 8px.
Like you have rules for transparency. Where you can set different level of transparency for different applications.

**Comment by alanpq** [comment]
Bumping this, is there any update on the feasibility of having this feature?

**Comment by absolutelynothelix** [comment]
it's now possible using the `corner-radius-rules` option (currently available only in the `next` branch because there was no picom release since it was added). thanks to @alanpq.

**Reply by voziand** [reply]
sweet!! big thanks!
